ale_linters/ruby/rubocop.vim

```diff
@@ -8,34 +8,51 @@ function! ale_linters#ruby#rubocop#GetCommand(buffer) abort
     \   : ''
 
     return ale#Escape(l:executable) . l:exec_args
-    \   . ' --format emacs --force-exclusion '
+    \   . ' --format json --force-exclusion '
     \   . ale#Var(a:buffer, 'ruby_rubocop_options')
     \   . ' --stdin ' . bufname(a:buffer)
 endfunction
 
 function! ale_linters#ruby#rubocop#Handle(buffer, lines) abort
-    " Matches patterns line the following:
-    "
-    " <path>:83:29: C: Prefer single-quoted strings when you don't
-    " need string interpolation or special symbols.
-    let l:pattern = '\v:(\d+):(\d+): (.): (.+)'
-    let l:output = []
+    if len(a:lines) == 0
+      return []
+    endif
+
+    let l:errors = json_decode(a:lines[0])
 
-    for l:match in ale#util#GetMatches(a:lines, l:pattern)
-        let l:text = l:match[4]
-        let l:type = l:match[3]
+    let l:output = []
 
+    for l:error in l:errors['files'][0]['offenses']
+        let l:start_col = l:error['location']['column'] + 0
         call add(l:output, {
-        \   'lnum': l:match[1] + 0,
-        \   'col': l:match[2] + 0,
-        \   'text': l:text,
-        \   'type': index(['F', 'E'], l:type) != -1 ? 'E' : 'W',
+        \   'lnum': l:error['location']['line'] + 0,
+        \   'col': l:start_col,
+        \   'end_col': l:start_col + l:error['location']['length'] - 1,
+        \   'text': l:error['message'],
+        \   'type': ale_linters#ruby#rubocop#GetType(l:error['severity']),
         \})
     endfor
 
     return l:output
 endfunction
 
+function! ale_linters#ruby#rubocop#GetType(severity) abort
+    if a:severity ==? 'refactor'
+      return 'W'
+    elseif a:severity ==? 'convention'
+      return 'W'
+    elseif a:severity ==? 'warning'
+      return 'W'
+    elseif a:severity ==? 'error'
+      return 'E'
+    elseif a:severity ==? 'fatal'
+      return 'E'
+    else
```

Comment:
Get rid of the message and return `'E'`. Then the worst case scenario is that you'll set some things as errors instead.

Also, you can write your code like this.

```vim
if ...
    return ...
endif

if ...
    return ...
endif

return ...
```